va/caa_test.go

```diff
@@ -647,3 +678,107 @@ func TestContainsMethod(t *testing.T) {
+		value          string
+		wantDomain     string
+		wantParameters map[string]string
+	}{
+		{
+			value:          "",
+			wantDomain:     "",
+			wantParameters: map[string]string{},
+		},
+		{
+			value:          ";",
+			wantDomain:     "",
+			wantParameters: map[string]string{},
+		},
+		{
+			value:          " ; ",
+			wantDomain:     "",
+			wantParameters: map[string]string{},
+		},
+		{
+			value:          "letsencrypt.org",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{},
+		},
+		{
+			value:          "letsencrypt.org;",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{},
+		},
+		{
+			value: "  letsencrypt.org	;foo=bar;baz=bar",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "bar", "baz": "bar"},
+		},
+		{
+			value: "	letsencrypt.org ;foo=bar;baz=bar",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "bar", "baz": "bar"},
+		},
+		{
+			value: "letsencrypt.org; foo=; baz =	bar",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "", "baz": "bar"},
+		},
+		{
+			value: "letsencrypt.org; foo=	; baz =	bar",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "", "baz": "bar"},
+		},
+		{
+			value: "letsencrypt.org; foo=b1,b2,b3	; baz =		a=b	",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "b1,b2,b3", "baz": "a=b"},
+		},
+		{
+			value: "letsencrypt.org; foo=b1,b2,b3	; baz =		a = b	",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "b1,b2,b3"},
+		},
+		{
+			value: "letsencrypt.org; foo=b1,b2,b3	; baz=a=	b",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "b1,b2,b3"},
+		},
+		{
+			value: "letsencrypt.org; foo=b1,b2,b3	; baz =		a;b	",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"foo": "b1,b2,b3", "baz": "a"},
+		},
+		{
+			value:          "letsencrypt.org; 1=2; baz=a-b",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"1": "2", "baz": "a-b"},
+		},
+		{
+			value:          "letsencrypt.org; 1=2; a-b=c",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{"1": "2"},
+		},
+		{
+			value:          "letsencrypt.org; foo=a\u2615b",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{},
+		},
+		{
+			value:          "letsencrypt.org; foo=b1,b2,b3 baz=a",
+			wantDomain:     "letsencrypt.org",
+			wantParameters: map[string]string{},
+		},
+	}
+	for _, test := range tests {
+		caa := &dns.CAA{Value: test.value}
+		gotDomain, gotParameters := extractIssuerDomainAndParameters(caa)
+		if got, want := gotDomain, test.wantDomain; got != want {
+			t.Errorf("CAA value %q - got domain %q, want %q", test.value, got, want)
+		}
+		if got, want := gotParameters, test.wantParameters; !reflect.DeepEqual(got, want) {
```

Comment:
This can be done with `test.AssertDeepEquals`.

Reply:
There are two downsides here - one `test.AssertDeepEquals` calls `t.Fatalf`, whereas this is using `t.Errorf` so that the group of values is visible, rather than fixing one failure, then finding the next. Secondly, `test.AssertDeepEquals` is not going to tell you want actually differs and just dump the two maps without any context. Please let me know if you still want this changed.

Reply:
I'm fine not using `test.AssertDeepEquals`. Might be nice to refactor this as a [golang style set of subtests](https://godoc.org/testing#hdr-Subtests_and_Sub_benchmarks).

Reply:
Done.